### DISCUSSION TITLE
Help setting up over 200 cameras

### DISCUSSION THREAD

**Discussion by Calin125** [blakeblackshear/frigate#15650] [Ask A Question]
Hi, 

I was wondering if you could help us with configuring over 200 cameras for our surveillance team. We are currently running 11 docker containers on a bare metal server with an Nvidia Tesla V100S. Each container is responsible for around 16-25 streams. I have configured ffmpeg to utilize the Nvidia GPU. However, the CPU usage is still high. We have built the app from source as we had to customize a couple of things. Also, is there any chance you could share the source code for the frigate:stable-tensorrt image? 

Here's a sample configuration I'm using for each instance. Each instance is running on a separate port i.e frigate 1 on port "8971" frigrate 2 on port "8972" and so on... I'm not sure if this is the optimal setup for our use case and hence if you could guide us on how to set it up. Thanks in advance. 

config.yml:

![Screenshot 2024-12-23 at 10 55 05](https://github.com/user-attachments/assets/5b6e8bb4-90cd-46e0-81af-b34fb7af671f)

docker-compose.yml

![Screenshot 2024-12-23 at 10 58 08](https://github.com/user-attachments/assets/33975202-7389-4c8d-ba38-a955fe6113ea)

**Comment by NickM-27** [comment]
Is there a reason why you have transcoding configured for every camera to h264? Also that transcoding is configured to be done on CPU not GPU

**Reply by Calin125** [reply]
Hi, thank you for your reply. That's strange, when I run nvidia-smi, it shows that ffmpeg is using the NVIDIA GPU. So I assumed that the transcoding is done by the GPU. I wasn't sure how to properly configure it so I just added h264 for every camera, but there's no specific reason for that, I should probably remove that line. I just need an optimal setup for live streaming without any lag. Performance is the main priority for our use case.  

Are you able to help me adjust the config.yml and docker-compose files for an optimal setup? 

Many thanks

**Reply by NickM-27** [reply]
If you don't have some reason to transcode to h264 then yes I'd suggest removing the line. After that we will need to see screenshots of the system and camera metrics so we can know what is using CPU

**Reply by Calin125** [reply]
Here's a screenshot of nvidia-smi.  Can you please let me know what system and camera metrics you would like to see. I'm not really sure how to get them. Can you please guide me on how to get them for you?

![MicrosoftTeams-image](https://github.com/user-attachments/assets/0d752799-b91e-4009-bbfa-94d00a938fdb)

**Reply by NickM-27** [reply]
`Settings Cog in bottom left` -> `System Metrics` and then a screenshot of that as well as the camera tab

**Reply by Calin125** [reply]
Here's a screenshot of the system metrics. Also, notice CPU usage is 56% compared to only 11% for the GPU. However, as I mentioned before, the server is running 11 containers and each container is running one app instance with 16-25 cameras each.

![Screenshot 2024-12-23 at 16 32 58](https://github.com/user-attachments/assets/5f1de71d-7bc4-4fe9-bd4e-3e6f9bbdb2b0)

**Reply by NickM-27** [reply]
Need to see the camera tab as well

**Comment by Calin125** [comment]
Sorry, I'm not sure what do you mean by sharing the camera tab? We have 16
live streams on there. I cannot share this atm.  Do you mean something
else?

Many thanks

**Reply by NickM-27** [reply]
I mean the camera tab of the metrics, in the top left, like I described in my previous comment

**Reply by Calin125** [reply]
I have shared it above. Are you not able to access the screenshot?

Here's the link again in case you're not able to access it.

![Screenshot 2024-12-23 at 16 32 58](https://github.com/user-attachments/assets/cfe6691e-2f4b-41b2-a84a-12cd7fa631fa)

**Reply by NickM-27** [reply]
you have shared the `General` tab, I am also asking for the `Camera` tab

**Reply by Calin125** [reply]
My bad, here's the camera metrics tab

![image](https://github.com/user-attachments/assets/740868a6-684e-4a19-9209-88edb9616c41)

**Reply by NickM-27** [reply]
I see you are only using one rtsp stream per camera. Is this truly the low res stream that is being used?

**Reply by Calin125** [reply]
These are the only rtsp links we were provided. The performance is not an issue, the stream is good even with the current rtsp links. I was just wondering if my config is an optimal solution for streaming around 200 cams.

**Reply by NickM-27** [reply]
Sure, but if you are decoding a high resolution stream and then resizing it to a low resolution this will use more CPU and is a waste, hence why I am asking.

**Reply by Calin125** [reply]
You're right, makes sense. I'll ask around and see if we can get the low resolutions links. Is there anything else, in your opinion we should do?

**Reply by NickM-27** [reply]
Not right now, most things seem about as optimal as they could be

**Reply by Calin125** [reply]
Ok, thank you for the feedback. Much appreciated. Just to confirm, I have removed the following lines in the go2rts streams:
ffmpeg:ar_investment_16#video=H264

Should I also remove the "h264" part from this line: hwaccel_args: preset-nvidia-h264

**Reply by NickM-27** [reply]
you can if you want to, but it works the same regardless
